### PR TITLE
fix: skip emitting private default provides.

### PR DESCRIPTION
**Pull request by rkirov**
Private default provides are not common, but when going through externs
we output each class and interface as a default provide.

### CODE REVIEW

File: src/main/java/com/google/javascript/clutz/DeclarationGenerator.java [hunk 1]
```diff
@@ -338,7 +338,12 @@ private int declareNamespace(String namespace, TypedVar symbol, boolean isDefaul
     emitNamespaceBegin(namespace);
     TreeWalker treeWalker = new TreeWalker(compiler.getTypeRegistry(), provides);
     if (isDefault) {
-      treeWalker.walk(symbol);
+      if (isPrivate(symbol.getJSDocInfo())) {
+        emit("// skipping emitting private default export " + symbol.getName() + ".");
```
Comment (mprobst) [inline src/main/java/com/google/javascript/clutz/DeclarationGenerator.java]: why is this a `default` export?

File: src/resources/third_party_externs.js [hunk 2]
```diff
@@ -127,4 +127,10 @@ ClassNamespace.constObj = {};
 /**
  * @type {number}
  */
-ClassNamespace.constObj.num;
\ No newline at end of file
+ClassNamespace.constObj.num;
```
Comment (mprobst) [inline src/resources/third_party_externs.js]: Please create a second test file for this, this is getting way too long. OK to do in a separate PR.

**Comment by mprobst** [comment]
LGTM.

File: src/main/java/com/google/javascript/clutz/DeclarationGenerator.java (hunk 1 above)
Comment (rkirov) [inline src/main/java/com/google/javascript/clutz/DeclarationGenerator.java]: as discussed offline, this is because we generate default module export for this symbol.

File: src/resources/third_party_externs.js (hunk 2 above)
Comment (rkirov) [inline src/resources/third_party_externs.js]: will need some new infrastructure. will do in another PR
